### PR TITLE
feat: complete paywall

### PR DESCRIPTION
**Pull request by kkortes**
Questions:
* Should we run `MailCheck` to suggest emails? (added bundle size)
* Should we check against `checkout.settings.fieldSettings` if phone, firstName & lastName ought to be asked for?
* Should we check "is already owner" (`checkout/<id>/owner`)?
* Should forms be native forms or is JS sufficient? (use `<form>` but don't allow for native action="url")

Todos:
* e2e-tests
* Make sure all old properties are supported and working
* Replace:
`https://api.sesamy.dev/checkout/<checkout-id>/owner`
`https://api.sesamy.dev/<checkout-id>/paymentMethod`
& call to `settings-url`

from sesamy-js

### CODE REVIEW

File: packages/lib/src/components/paywall/PayNowForm.svelte [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  import PaymentMethod from '../PaymentMethod.svelte';
+  import type { IconName } from '../../icons/types';
+  import Icon from '../Icon.svelte';
+  import Row from '../Row.svelte';
+  import { isValidEmail } from '../../utils/email';
+  import Error from '../Error.svelte';
+  import Column from '../Column.svelte';
+
+  // TODO: can we these dynamic (.com if built for prod, .dev if ran on local/preview)
+  const API_URL = 'https://api.sesamy.dev';
```
[the remark this answers is not in the document]
Comment (kkortes) [reply packages/lib/src/components/paywall/PayNowForm.svelte:20]: Then I'll these two:
`https://api.sesamy.dev/checkout/<checkout-id>/owner`
`https://api.sesamy.dev/<checkout-id>/paymentMethod`

I'm guessing the network call in `Paywall.wc.svelte` still should be made from the component itself (and not Sesamy-js?). It's the settings-url to-grab-json one

File: packages/lib/src/components/paywall/PayNowForm.svelte [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+              method,
+              icon: method.toLocaleLowerCase() as IconName
+            }))
+          : [{ provider, method: undefined, icon: provider.toLocaleLowerCase() as IconName }])
+      ],
+      [] as PaymentMethod[]
+    )
+    .filter(({ method }) => method !== 'SWISH');
+
+  /Chrome/.test(navigator?.userAgent) &&
```
[the remark this answers is not in the document]
Comment (kkortes) [reply packages/lib/src/components/paywall/PayNowForm.svelte]: Done.

File: packages/lib/src/components/paywall/PayNowForm.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (markusahlstrand) [reply packages/lib/src/components/paywall/PayNowForm.svelte:21]: There's a patch endpoint. Not sure if I added it to sesamy-js? I think we should fetch the paywall-json via sesamy-js as well, but I need to add some plumming for that first.

[the remark this answers is not in the document]
Comment (markusahlstrand) [reply packages/lib/src/components/paywall/PayNowForm.svelte:21]: Yeah, seems to be in there. It takes the same object as the create but you just patch it with the the partial object. I'll check that support the payment method and the owner

[the remark this answers is not in the document]
Comment (kkortes) [reply packages/lib/src/components/paywall/PayNowForm.svelte:21]: I'll get back to this on monday. 

Would that be `api.checkouts.update`? there is no patch only `update`, `create` and `get`

**Review by markusahlstrand — APPROVED** [review]
This looks fine to me

**Comment by kkortes** [comment]
> This looks fine to me

I was about to merge this after generating new snapshots. But I cannot generate snapshots from within the docker image.

Let's chat this through on Monday. I'm suspecting you generated snapshots outside the docker container, which isn't something that can/should be done.

**Pull request merged**